apps/api/v2/src/ee/calendars/services/ics-feed.service.ts

```diff
@@ -0,0 +1,116 @@
+import { ICSFeedCalendarApp } from "@/ee/calendars/calendars.interface";
+import { CreateIcsFeedOutputResponseDto } from "@/ee/calendars/input/create-ics.output";
+import { CalendarsService } from "@/ee/calendars/services/calendars.service";
+import { CredentialsRepository } from "@/modules/credentials/credentials.repository";
+import { BadRequestException, UnauthorizedException, Logger } from "@nestjs/common";
+import { Injectable } from "@nestjs/common";
+
+import { SUCCESS_STATUS, ICS_CALENDAR_TYPE, ICS_CALENDAR } from "@calcom/platform-constants";
+import { symmetricEncrypt, IcsFeedCalendarService } from "@calcom/platform-libraries";
+
+@Injectable()
+export class IcsFeedService implements ICSFeedCalendarApp {
+  constructor(
+    private readonly calendarsService: CalendarsService,
+    private readonly credentialRepository: CredentialsRepository
+  ) {}
+
+  private logger = new Logger("IcsFeedService");
+
+  async save(
+    userId: number,
+    userEmail: string,
```

Comment:
Can we make `save` accept either "userId" or "userEmail" but not both, and then if "userId" is passed we fetch the user and then pass down the user email manually - all this is to simplify developer experience.

Reply:
we already fetch the user via @GetUser i'd rather not do it twice